pkg/operator/latencyprofilecontroller/controller.go

```diff
@@ -0,0 +1,216 @@
+}
+
+func (c *LatencyProfileController) sync(ctx context.Context, syncCtx factory.SyncContext) error {
+	syncMethod := c.ensureProfileSyncedToAllNodes
+	err := syncMethod(ctx, syncCtx)
+	utilruntime.HandleError(err)
+	return err
+}
+
+// ensureProfileSyncedToAllNodes
+func (c *LatencyProfileController) ensureProfileSyncedToAllNodes(ctx context.Context, syncCtx factory.SyncContext) error {
+	_, operatorStatus, _, err := c.operatorClient.GetStaticPodOperatorState()
+	if err != nil {
+		return err
+	}
+
+	// Collect the unique set of revisions of the apiserver nodes
+	revisionMap := map[int32]struct{}{}
+	uniqueRevisions := []int32{}
+	for _, nodeStatus := range operatorStatus.NodeStatuses {
+		revision := nodeStatus.CurrentRevision
+		if _, ok := revisionMap[revision]; !ok {
+			revisionMap[revision] = struct{}{}
+			uniqueRevisions = append(uniqueRevisions, revision)
+		}
+	}
+
+	// Collect the current latency profile
+	desiredApiServerArgumentVals := map[string]string{}
+	configNodeObj, err := c.nodeLister.Get("cluster")
+	if err != nil {
+		return err
+	}
+
+	switch {
+	case configNodeObj.Spec.WorkerLatencyProfile == configv1.DefaultUpdateDefaultReaction:
+		desiredApiServerArgumentVals[DefaultNotReadyTolerationSecondsArgument] = strconv.Itoa(configv1.DefaultNotReadyTolerationSeconds)
+		desiredApiServerArgumentVals[DefaultUnreachableTolerationSecondsArgument] = strconv.Itoa(configv1.DefaultUnreachableTolerationSeconds)
+	case configNodeObj.Spec.WorkerLatencyProfile == configv1.DefaultUpdateDefaultReaction:
+		desiredApiServerArgumentVals[DefaultNotReadyTolerationSecondsArgument] = strconv.Itoa(configv1.MediumNotReadyTolerationSeconds)
+		desiredApiServerArgumentVals[DefaultUnreachableTolerationSecondsArgument] = strconv.Itoa(configv1.MediumUnreachableTolerationSeconds)
+	case configNodeObj.Spec.WorkerLatencyProfile == configv1.DefaultUpdateDefaultReaction:
+		desiredApiServerArgumentVals[DefaultNotReadyTolerationSecondsArgument] = strconv.Itoa(configv1.LowNotReadyTolerationSeconds)
+		desiredApiServerArgumentVals[DefaultUnreachableTolerationSecondsArgument] = strconv.Itoa(configv1.LowUnreachableTolerationSeconds)
+	}
+	// do nothing if workerLatencyProfile is not set
+	if configNodeObj.Spec.WorkerLatencyProfile == "" {
+		// TODO: in case worker latency profile is transitioned
+		// from Default/Medium/Low to ""
+		// check that the required apiServerArgs should vanish
+		newCondition := configv1.WorkerLatencyStatusCondition{
+			Owner:              configv1.KubeAPIServerOperator,
+			Status:             configv1.ConditionUnknown,
+			LastTransitionTime: metav1.Now(),
+			Message:            "Worker latency profile not set",
+		}
+		c.updateConfigNodeStatus(configNodeObj, &newCondition)
+	}
+
+	// For each revision, check that the configmap for that revision have
+	// correct argument values or not
+	revisionsHaveSynced := true
+	for _, revision := range uniqueRevisions {
```

Comment:
shouldn't we check only a single revision? 

I think we should wait until all API servers are on the same revision and then simply check the latest.

Reply:
While implementing this:
- took majority of my inspiration from: 

https://github.com/openshift/cluster-kube-apiserver-operator/blob/10569df18e9577663802c0df9861bce540b16e2b/pkg/operator/boundsatokensignercontroller/controller.go#L238

https://github.com/openshift/cluster-kube-apiserver-operator/blob/10569df18e9577663802c0df9861bce540b16e2b/pkg/operator/boundsatokensignercontroller/controller.go#L257-L292

- this should be able to take care of the fact that more/less irrespective of different revision rollout(s) by various config observers the worker latency profile status stays up to date via checks on the given node revision if the required apiServerArgument is present or not
- also: (https://coreos.slack.com/archives/CB48XQ4KZ/p1648141150207779?thread_ts=1648139872.258959&cid=CB48XQ4KZ)
> > In case the KubeAPIServer resource on the cluster is changed twice within a short interval of time (assuming the changes were done by config observers), by design: will the operator's revision control pods transition the kube-apiserver static pods from revision x directly to revision x+2 or attempt the revisions consecutively i.e. revision x+1, revision x+2?
>
> It depends on timing and your luck, but I think it is possible to go straight to x+2.
At worst (assuming revision creation interval < time to apply revision to a single node), only one node will get x+1 and then x+2 , subsequent nodes will go directly to x+2. 

- so, in case of n frequent rollout(s) by different config observers nodes could attain revision x+n directly instead of x+1, x+2, ... x+n eventually; in case we wait for all nodes to attain target revision we the controller would keep waiting / take very long / never attain profile rollout Complete status.